test/framework/e2e_framework.go

```diff
@@ -56,7 +56,7 @@ var (
 	nodeToolStatusDN = regexp.MustCompile(`DN\s\s`)
 )
 
-func NewE2eFramework(t *testing.T, kubeconfigFile string, useDse bool, controlPlane string, dataPlanes ...string) (*E2eFramework, error) {
+func NewE2eFramework(t *testing.T, kubeconfigFile string, useDse bool, useHcd bool, controlPlane string, dataPlanes ...string) (*E2eFramework, error) {
```

Comment:
issue: This isn't ideal, I'd like to see serverType used directly here since setting both useHcd and useDse shouldn't be set at the same time.